lib/auth/clt.go

```diff
@@ -137,9 +139,23 @@ func NewHTTPClient(cfg client.Config, tls *tls.Config, params ...roundtrip.Clien
 		dialer = client.ContextDialerFunc(func(ctx context.Context, network, _ string) (conn net.Conn, err error) {
 			for _, addr := range cfg.Addrs {
 				conn, err = contextDialer.DialContext(ctx, network, addr)
-				if err == nil {
-					return conn, nil
+				if err != nil {
+					continue
 				}
+				if source, ok := ctx.Value(utils2.ContextClientAddr).(*net.TCPAddr); ok {
+					line := multiplexer.ProxyLine{
```

Comment:
I think it's a bit weird we would use PROXY protocol for propagating the client IP when making requests to the HTTP API, but what is essentially a header ( `metadata` ) when gRPC requests . I would probably strive to use either headers/metadata for both, or PROXY protocol for both.

My gut feeling is to align on headers/metadata rather than the PROXY protocol. PROXY protocol gives me a few concerns:
- A malicious actor can spoof their IP address if the Auth server has PROXY protocol enabled, and the user can access it without passing through a load-balancer. 
- The user could disable PROXY protocol, in which case this functionality would stop working. We would need to come up with some solution for a special PROXY protocol which is always enabled, and then establish that this had come from a genuine Proxy. 
- PROXY protocol seems liable to issues if a connection is re-used. This seems theoretically possible, since an attacker could steal a users Proxy web API session cookie from their browser, and the Proxy web API caches `auth.Client` based on the user ID and session ID. This means the the `auth.Client` would be reused, and its possible that this could reuse a previous connection where the genuine users IP address has been sent via the PROXY protocol.